internal/helpers/config.go

```diff
@@ -48,11 +47,9 @@ func GetConfigBool(ctx context.Context, configValue basetypes.BoolValue, environ
 		envValue, err := strconv.ParseBool(value)
 		if err == nil {
 			return envValue
-		} else {
-			tflog.Warn(ctx, "Failed to parse environment variable value as a boolean. Using default value instead.", map[string]interface{}{environmentVariableName:value})
-			return defaultValue
 		}
-	} else {
+		tflog.Warn(ctx, "Failed to parse environment variable value as a boolean. Using default value instead.", map[string]any{environmentVariableName: value})
 		return defaultValue
```

Comment:
This feels backwards to me.  Consider handling the err != nil condition in the if and returning envValue after the if